src/data/weather.ts

```diff
@@ -538,3 +544,49 @@ export const getWeatherConvertibleUnits = (
   hass.callWS({
     type: "weather/convertible_units",
   });
+
+export const getForecast = (
```

Comment:
This function should take `this.stateObj.attributes` as param.

Reply:
And return the forecast + it's properties:
```ts
{forecast: ForecastAttribute[], type: "daily" | "hourly" | "twice_daily"};
```
So for legacy, we will move the `isForecastHourly` and `dayNight` checks here.